bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/handler/SoulissT41Handler.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.souliss.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.souliss.SoulissBindingConstants;
+import org.openhab.binding.souliss.SoulissBindingProtocolConstants;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.PrimitiveType;
+import org.openhab.core.types.RefreshType;
+
+/**
+ * The {@link SoulissT41Handler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Luca Remigio - Initial contribution
+ * @author Luca Calcaterra - Refactor for OH3
+ */
+
+@NonNullByDefault
+public class SoulissT41Handler extends SoulissGenericHandler {
+
+    @Nullable
+    Configuration gwConfigurationMap;
+    byte t4nRawState;
+
+    public SoulissT41Handler(Thing thing) {
+        super(thing);
+    }
+
+    // called on every status change or change request
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!(command instanceof RefreshType)) {
+            if (channelUID.getAsString().split(":")[3].equals(SoulissBindingConstants.T4N_ONOFFALARM_CHANNEL)) {
+                if (command instanceof OnOffType) {
+                    switch (command.toFullString()) {
+                        case "OFF":
+                            commandSEND(SoulissBindingProtocolConstants.SOULISS_T4N_NOT_ARMED);
+                            break;
+                        case "ON":
+                            commandSEND(SoulissBindingProtocolConstants.SOULISS_T4N_ARMED);
+                            break;
+                    }
```

Comment:
```suggestion
                    switch (command) {
                        case OnOffType.OFF:
                            commandSEND(SoulissBindingProtocolConstants.SOULISS_T4N_NOT_ARMED);
                            break;
                        case OnOffType.ON:
                            commandSEND(SoulissBindingProtocolConstants.SOULISS_T4N_ARMED);
                            break;
                    }
```

More efficient if I got the syntax right.

Reply:
your edit not working on switch case statement , error on command type (only int or primitives)

Reply:
Sorry, the syntax was wrong as I did not try in an IDE, then use ....

if(command.equals(OnOffType.OFF)){
commandSEND(SoulissBindingProtocolConstants.SOULISS_T4N_NOT_ARMED);
}else if(command.equals(OnOffType.ON)){
commandSEND(SoulissBindingProtocolConstants.SOULISS_T4N_ARMED);
}

It should be more efficient due to not having to convert to a String and compare as Strings.

Reply:
fixed